### PR TITLE
pass file info to rules

### LINKED ISSUE #30: Access current file name/path in custom rule
Is there some way to access the name/path for the current file in a custom rule? It doesn't seem to be available on any of the arguments passed to the function.

### PR DESCRIPTION
**Pull request by WickyNilliams**
fixes #30

### CODE REVIEW

**Comment by WickyNilliams** [comment]
Sorry it took so long 😄

File: src/rules/custom.js
Comment (ericcornelissen) [inline src/rules/custom.js:31]: If it's intended purely for information about the file being linted, would it make more sense to call this parameter `file` (and the type `File`)?

Else, if the intend is to keep it generic, should the object type definition be changed to, e.g.:

```js
/** @typedef {{ file: { path: string } }} Info */
```

or

```js
/** @typedef {{ filepath: string }} Info */
```

**Review by ericcornelissen — COMMENTED** [review]
> Sorry it took so long :smile:

No worries, better late then never :smiley: 

The implementation looks good. Just have one comment to potentially improve clarity and/or extensibility. Also, as [mentioned in the issue](https://github.com/birjolaxew/svglint/issues/30#issuecomment-857744379), this is ideally tested as well. I'm a new maintainer and not too familiar with the test suite for the project, but I think a test like the following in [`custom.spec.js`](https://github.com/birjolaxew/svglint/blob/9aee2df593ee7fcee199345bf48491c550a50c1e/test/custom.spec.js) could be a start:

```js
it("should provide file information", function(done){
        testSucceeds([
            (reporter, $, ast, info) => {
                expect(info).toExist();
                expect(info.hasOwnProperty("path")).toBe(true);
                done();
            }
        ]);
    });
```

File: src/rules/custom.js
Comment (WickyNilliams) [reply src/rules/custom.js:31]: renamed path -> filepath

**Comment by WickyNilliams** [comment]
I added a test as you asked. 

I struggled to get the test to work properly (it would dump a load of junk to the console) using the format you suggested, so instead i throw an Error if either info, or info.filepath or not present. Hope that's ok!

File: test/custom.spec.js
Comment (ericcornelissen) [inline test/custom.spec.js:102]:
```suggestion
    it("should provide file information", function(){
```

**Review by ericcornelissen — CHANGES_REQUESTED** [review]
> I added a test as you asked.

Thanks!

> I struggled to get the test to work properly (it would dump a load of junk to the console) using the format you suggested, so instead i throw an Error if either `info`, or `info.filepath` is not present. Hope that's ok!

Apologies if my test didn't work for you. Anyway, I do quite like the way you did it so no problems there.

Could you just remove the use of `.only` from and [add semicolons `;` where they're missing](https://github.com/birjolaxew/svglint/runs/4131592601?check_suite_focus=true#step:5:10) in the test suite?

File: test/custom.spec.js
Comment (WickyNilliams) [reply test/custom.spec.js]: leaving `only` on tests is something of a speciality of mine haha. i do it _all_ the time. i've fixed now, and squashed into the last commit, to avoid some noise in the commit history

**Review by ericcornelissen — APPROVED** [review]
This looks good to me now @WickyNilliams, thanks!

~~I'll leave this open for a few more days (at least until 2021-11-15) to give @birjolaxew a chance to review it as well.~~

**Review by birjolaxew — APPROVED** [review]

**Comment by birjolaxew** [comment]
Haven't had the chance to actually test, but the code looks good to me. I'll let @ericcornelissen merge.

**Pull request merged**
